### PR TITLE
Adding usaddress and dependencies

### PR DESCRIPTION
**Pull request by wyseguy7**
This is my first conda recipe submission, apologies for mistakes

### CODE REVIEW

**Comment by ericdill** [comment]
@ocefpaf any idea how to fix this windows error?

```
C:\Program Files (x86)\Windows Kits\10\include\10.0.10586.0\ucrt\stdio.h(1927): fatal error C1189: #error:  Macro definition of snprintf conflicts with Standard Library function declaration
```

https://ci.appveyor.com/project/conda-forge/staged-recipes/build/1.0.4481/job/qxvvwx7uq4w6w6jk#L686

**Comment by ocefpaf** [comment]
Sorry @ericdill. Windows is not my forte. But since it is failing only on PY35 I guess it is a `vc` thing. Maybe @patricksnape or @gillins have some idea of what may be going on.

**Comment by jakirkham** [comment]
I think this SO [answer]( http://stackoverflow.com/a/27754829 ) captures the issue. Though I'm not a Windows expert either.

**Comment by jakirkham** [comment]
Maybe it is worth breaking up this into two PRs. That way we can clean up and clear the working ones and ping those who know Windows better to take a look at the non-working ones.

**Comment by patricksnape** [comment]
Yup, `crfsuite` is doing some naughty stuff. See [here](https://github.com/chokkan/crfsuite/blob/387ca696e698d2d2865337acb1ee52a46b086749/include/os.h#L50). Simplest thing is to patch the `_MSC_VER` def in order to only apply for `_MSC_VER<1900` and to add the `STDC99` define.

**Comment by jakirkham** [comment]
The Travis CI job is queued for this. It is just being a bit slow to update the status for the PR today. 😒

File: recipes/metafone/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    run:
+        - python
+
+test:
+    imports:
+        - metaphone
+
+about:
+    home: https://github.com/al45tair/metafone
+    license: BSD
```
Comment (jakirkham) [inline recipes/metafone/meta.yaml]: Please specify number of clauses.

File: recipes/metafone/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        - future
+    run:
+        - python
+
+test:
+    imports:
+        - metaphone
+
+about:
+    home: https://github.com/al45tair/metafone
```
Comment (jakirkham) [inline recipes/metafone/meta.yaml]: This URL doesn't seem to work. 😕

Comment (jakirkham) [inline recipes/metafone/meta.yaml]: PyPI put this as https://github.com/oubiwann/metaphone

File: recipes/metafone/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+build:
+    number: 0
+    script: python setup.py install --single-version-externally-managed --record record.txt
+
+requirements:
+    build:
+        - python
+        - setuptools
+        - future
```
Comment (jakirkham) [inline recipes/metafone/meta.yaml]: Why is `future` only a build dependency? Shouldn't it be a run dependency too?

File: recipes/probableparsing/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+build:
+    number: 0
+    script: python setup.py install --single-version-externally-managed --record record.txt
+
+requirements:
+    build:
+        - python
+        - setuptools
+        - future
```
Comment (jakirkham) [inline recipes/probableparsing/meta.yaml]: Why is `future` only a build dependency? Shouldn't it be a run dependency too?

**Comment by jakirkham** [comment]
`pysal` is already removed. So please rebase to get that out of this PR.

**Comment by jakirkham** [comment]
Does `python-crfsuite` have a test suite we can run (since it is compiled)?

File: recipes/python-crfsuite/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  {{ hashtype }}: {{ hash }}
+
+build:
+    number: 0
+    script: python setup.py install --single-version-externally-managed --record record.txt
+    skip: True  # [win and py35]
+
+requirements:
+    build:
```
Comment (jakirkham) [inline recipes/python-crfsuite/meta.yaml]: Please add `toolchain` to the dependencies.

File: recipes/python-crfsuite/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  {{ hashtype }}: {{ hash }}
+
+build:
+    number: 0
+    script: python setup.py install --single-version-externally-managed --record record.txt
```
Comment (jakirkham) [inline recipes/python-crfsuite/meta.yaml]: Please change to...

```
  script:
    # FIXME: This is a hack to make sure the environment is activated.
    # The reason this is required is due to the conda-build issue
    # mentioned below.
    #
    # https://github.com/conda/conda-build/issues/910
    #
    - source activate "${CONDA_DEFAULT_ENV}"  # [unix]
```

Comment (wyseguy7) [inline recipes/python-crfsuite/meta.yaml]: You want me to add this to each recipe or just this one?

**Comment by wyseguy7** [comment]
@jakirkham I think i've addressed all of your comments, does this look sufficient?

File: recipes/probableparsing/meta.yaml (hunk 4 above)
Comment (jakirkham) [inline recipes/probableparsing/meta.yaml]: Done.

File: recipes/probableparsing/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+    run:
+        - python
+        - future
+test:
+    imports:
+        - {{name}}
```
Comment (jakirkham) [inline recipes/probableparsing/meta.yaml]: We should probably just make this `probableparsing`, but it can be done here or saved for the feedstock.

**Comment by jakirkham** [comment]
Yep, looks reasonable to me. One comment, but it can be addressed later if you wish.

**Comment by wyseguy7** [comment]
Yeah, let's address it later. Just to clarify, you're suggesting we split probableparsing out into a separate repo?

**Comment by jakirkham** [comment]
> Just to clarify, you're suggesting we split probableparsing out into a separate repo? 

Sorry not quite following. Are you asking about what the comment means or when we should address it? If the former, I'm just suggesting a trivial line edit. If the later, I'm saying it can be done after we merge.

**Comment by wyseguy7** [comment]
Oh, sorry. I get it now. Let's just merge and do the edit later.

**Comment by jakirkham** [comment]
Thanks @wyseguy7 for the contributions.
 
Also thanks @patricksnape for tracking down the Windows issue. Would suggest the maintainers refer this issue upstream and consider applying a simple Windows patch along these lines.

Thanks everyone else for helping with the review process.

**Comment by jakirkham** [comment]
> Oh, sorry. I get it now.

No worries.

> Let's just merge and do the edit later.

👍 Done. 😄

File: recipes/probableparsing/meta.yaml (hunk 7 above)
Comment (jakirkham) [inline recipes/probableparsing/meta.yaml]: I added a PR ( https://github.com/conda-forge/probableparsing-feedstock/pull/1 ) to make this change.

**Pull request merged**